### PR TITLE
[SYCL][ESIMD] Lower globals to vector in opaque pointer mode

### PR DESCRIPTION
**Pull request by sarnex**
We only did this in ESIMDLowerVecArg which is not run when opaque pointers is on.

We always need this to happen, so move it to common code.

### CODE REVIEW

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp [hunk 1]
```diff
@@ -1775,6 +1775,70 @@ void lowerGlobalStores(Module &M, const SmallPtrSetImpl<Type *> &GVTS) {
...
+    Type *NewTy = esimd::getVectorTyOrNull(dyn_cast<StructType>(GVTy));
+    if (NewTy && !G.user_empty()) {
+      auto InitVal =
+          G.hasInitializer() && isa<UndefValue>(G.getInitializer())
+              ? static_cast<ConstantData *>(UndefValue::get(NewTy))
+              : static_cast<ConstantData *>(ConstantAggregateZero::get(NewTy));
+      auto NewGlobalVar =
+          new GlobalVariable(NewTy, G.isConstant(), G.getLinkage(), InitVal, "",
+                             G.getThreadLocalMode(), G.getAddressSpace());
+      NewGlobalVar->setExternallyInitialized(G.isExternallyInitialized());
```
Comment (turinevgeny) [inline llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:1823]: is copying Visibility also important? (`void llvm::GlobalValue::setVisibility(VisibilityTypes V)`)
I see that this is not a new code, but just wondering.

**Review by turinevgeny — APPROVED** [review]
LGTM, only a small question.

File: llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp (hunk 1 above)
Comment (turinevgeny) [reply llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp:1823]: Ok, thanks!

**Comment by sarnex** [comment]
@intel/llvm-gatekeepers we are good to merge, thx!

**Pull request merged**